### ISSUE TITLE
Busybox fails to build when nc with extra options is selected

### ISSUE THREAD

**Issue by bclau** [rmyorston/busybox-w32#117]
Building Busybox with the default options enabled works just fine, but when enabling some extra option (nc related), the following errors occur:

- when enabling Netcat 1.10 compatibility (+2.5k):

    libbb/udp_io.c: In function Ã¢nd_to_fromÃ¢libbb/udp_io.c:41:15: error: array type has incomplete element type Ã¢ruct iovecÃ¢ struct iovec iov[1];
                   ^
    libbb/udp_io.c:42:16: error: storage size of Ã¢gÃ¢snÃ¢known
      struct msghdr msg;
                    ^
    libbb/udp_io.c:44:13: warning: implicit declaration of function Ã¢SG_SPACEÃ¢-Wimplicit-function-declaration]
       char cmsg[CMSG_SPACE(sizeof(struct in_pktinfo))];
                 ^
    libbb/udp_io.c:76:12: warning: implicit declaration of function Ã¢SG_FIRSTHDRÃ¢-Wimplicit-function-declaration]
      cmsgptr = CMSG_FIRSTHDR(&msg);
                ^
    libbb/udp_io.c:85:10: error: dereferencing pointer to incomplete type Ã¢ruct cmsghdrÃ¢  cmsgptr->cmsg_level = IPPROTO_IP;
              ^
    libbb/udp_io.c:87:23: warning: implicit declaration of function Ã¢SG_LENÃ¢-Wimplicit-function-declaration]
       cmsgptr->cmsg_len = CMSG_LEN(sizeof(struct in_pktinfo));
                           ^
    In file included from /usr/share/mingw-w64/include/windows.h:95:0,
                     from /usr/share/mingw-w64/include/winsock2.h:23,
                     from include/libbb.h:146,
                     from libbb/udp_io.c:9:
    libbb/udp_io.c:88:34: error: called object is not a function or function pointer
       pktptr = (struct in_pktinfo *)(CMSG_DATA(cmsgptr));
                                      ^
    libbb/udp_io.c:96:9: error: Ã¢ruct in_pktinfoÃ¢as no member named Ã¢i_spec_dstÃ¢  pktptr->ipi_spec_dst = ((struct sockaddr_in*)from)->sin_addr;
             ^
    In file included from /usr/share/mingw-w64/include/windows.h:95:0,
                     from /usr/share/mingw-w64/include/winsock2.h:23,
                     from include/libbb.h:146,
                     from libbb/udp_io.c:9:
    libbb/udp_io.c:104:35: error: called object is not a function or function pointer
       pktptr = (struct in6_pktinfo *)(CMSG_DATA(cmsgptr));
                                       ^
    libbb/udp_io.c:111:9: warning: implicit declaration of function Ã¢ndmsgÃ¢-Wimplicit-function-declaration]
      return sendmsg(fd, &msg, flags);
             ^
    libbb/udp_io.c:42:16: warning: unused variable Ã¢gÃ¢-Wunused-variable]
      struct msghdr msg;
                    ^
    libbb/udp_io.c:41:15: warning: unused variable Ã¢vÃ¢-Wunused-variable]
      struct iovec iov[1];
                   ^
    libbb/udp_io.c: In function Ã¢cv_from_toÃ¢libbb/udp_io.c:129:15: error: array type has incomplete element type Ã¢ruct iovecÃ¢ struct iovec iov[1];
                   ^
    libbb/udp_io.c:137:16: error: storage size of Ã¢gÃ¢snÃ¢known
      struct msghdr msg;
                    ^
    libbb/udp_io.c:151:16: warning: implicit declaration of function Ã¢cvmsgÃ¢-Wimplicit-function-declaration]
      recv_length = recvmsg(fd, &msg, flags);
                    ^
    libbb/udp_io.c:160:14: warning: implicit declaration of function Ã¢SG_NXTHDRÃ¢-Wimplicit-function-declaration]
        cmsgptr = CMSG_NXTHDR(&msg, cmsgptr)
                  ^
    libbb/udp_io.c:162:14: error: dereferencing pointer to incomplete type Ã¢ruct cmsghdrÃ¢  if (cmsgptr->cmsg_level == IPPROTO_IP
                  ^
    In file included from /usr/share/mingw-w64/include/windows.h:95:0,
                     from /usr/share/mingw-w64/include/winsock2.h:23,
                     from include/libbb.h:146,
                     from libbb/udp_io.c:9:
    libbb/udp_io.c:169:35: error: called object is not a function or function pointer
        memcpy(&to4->sin_addr, (char*)(CMSG_DATA(cmsgptr)) + IPI_ADDR_OFF, sizeof(to4->sin_addr));
                                       ^
    libbb/udp_io.c:181:36: error: called object is not a function or function pointer
        memcpy(&to6->sin6_addr, (char*)(CMSG_DATA(cmsgptr)) + IPI6_ADDR_OFF, sizeof(to6->sin6_addr));
                                        ^
    libbb/udp_io.c:137:16: warning: unused variable Ã¢gÃ¢-Wunused-variable]
      struct msghdr msg;
                    ^
    libbb/udp_io.c:129:15: warning: unused variable Ã¢vÃ¢-Wunused-variable]
      struct iovec iov[1];
                   ^
    libbb/udp_io.c: In function Ã¢nd_to_fromÃ¢libbb/udp_io.c:113:1: warning: control reaches end of non-void function [-Wreturn-type]
     }

- when enabling Netcat extensions (-eiw and -f FILE)

    miscutils/man.c: In function Ã¢d_MANPATHÃ¢miscutils/man.c:234:14: warning: Ã¢veÃ¢ay be used uninitialized in this function [-Wmaybe-uninitialized]
       *next_path = save;
                  ^
      CC      miscutils/strings.o
      CC      miscutils/ttysize.o
      AR      miscutils/lib.a
      CC      networking/ftpgetput.o
      CC      networking/ipcalc.o
      CC      networking/nc.o
    networking/nc.c: In function Ã¢_mainÃ¢networking/nc.c:194:10: error: Ã¢GALRMÃ¢ndeclared (first use in this function)
       signal(SIGALRM, timeout);
              ^
    networking/nc.c:194:10: note: each undeclared identifier is reported only once for each function it appears in
    networking/nc.c:241:11: error: Ã¢GCHLDÃ¢ndeclared (first use in this function)
        signal(SIGCHLD, SIG_IGN);
               ^
    scripts/Makefile.build:197: recipe for target 'networking/nc.o' failed
    make[1]: *** [networking/nc.o] Error 1
    Makefile:747: recipe for target 'networking' failed
    make: *** [networking] Error 2

OS: Ubuntu 16.04 (uname -a: Linux ubuntu 4.4.0-87-generic #110-Ubuntu SMP Tue Jul 18 12:55:35 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux)

**Comment by rmyorston** [comment]
Yes, if something isn't enabled in the default configuration it's probably because it doesn't work.

BusyBox uses various features from Unix, Linux and POSIX which aren't available in Microsoft Windows. Replacements for some of these have been implemented in busybox-w32, but many haven't.

In this case we're missing support for:
- ```sendmsg```/```recvmsg``` for UDP communication
- the ```SIGALRM``` and ```SIGCHLD``` signals

There aren't currently any plans to implement these.
